In [1]:
import mne
import os
import autoreject as aj
import numpy as np
import pandas as pd
from pathlib import Path

DATA_PATH = Path("../data/")

In [239]:
path = DATA_PATH / "post_ica"
out_path = DATA_PATH / "epoched"
subj_names = []

for file in path.glob("*"):
    if 'raw.fif' in file.name:
        subj_names.append(file.name.split('_')[2])

print(subj_names)

['VP10', 'VP11', 'VP12', 'VP13', 'VP14', 'VP15', 'VP16', 'VP17', 'VP18', 'VP19', 'VP1', 'VP20', 'VP21', 'VP22', 'VP23', 'VP24', 'VP2', 'VP3', 'VP5', 'VP6', 'VP7', 'VP8', 'VP9']


In [57]:
# Merging the same events to a single sample number
def merge_events(events):
    events = mne.merge_events(events, [11,12,20,21,22,23,31,32], 11, replace_events=True) # stimulus/t1
    events = mne.merge_events(events, [64,65,66,67,68,128,129,130,131,132,133], 64, replace_events=True) # stimulus/t2
    events = mne.merge_events(events, [1,2,3], 1, replace_events=True) # response/t1
    events = mne.merge_events(events, [4,5], 4, replace_events=True) # response/t2
    events = mne.merge_events(events, [6,7], 6, replace_events=True) # block

    return events

In [58]:
# Decide if trial has a short or long SOA based on time difference of stimulus onsets
def find_condition(events):
    new_events = []
    existing_event_onsets_set1 = np.concatenate([events[events[:, 2] == 11, 0]]) # stimulus/t1
    existing_event_onsets_set2 = np.concatenate([events[events[:, 2] == 64, 0]]) # stimulus/t2
    
    for idx, x in enumerate(existing_event_onsets_set1):     
        # The time difference between the two stimuli is computed by finding for every stim1 the closest positive difference with all stim2
        diff = existing_event_onsets_set2 - x
        valid_indices = np.where(diff > 0)
        closest_index = np.argmin(diff[valid_indices])

        # Prevents error if there are more stimulus 1 events than stimulus 2 events
        if idx == len(existing_event_onsets_set2): break
        
        if diff[valid_indices][closest_index] > 600 and diff[valid_indices][closest_index] < 615: # to allow for some variation in the samples
            new_events.append([x, 0, 256])  # long delay
        elif diff[valid_indices][closest_index] > 140 and diff[valid_indices][closest_index] < 151:
            new_events.append([x, 0, 512])  # short delay
        else:
            new_events.append([x, 0, 1024])  # exclude
    return new_events

In [59]:
def get_events(raw, tmin, tmax):
    stimulus_id = {'stimulus/t1': 11, 'stimulus/t2': 64} # Stimulus trigger
    condition_id = {"condition/short": 512, "condition/long": 256} # Condition trigger
    response_id = {'response/t2': 4, 'response/t1': 1} # Response triggers
    event_id = condition_id | stimulus_id | response_id # All retained events

    events = mne.find_events(raw, shortest_event=1)
    events = merge_events(events)

    # Identify new events across trigger sets and consider only the two nearest events
    new_events = find_condition(events)
    new_events = np.array(new_events)
    events_temp = np.concatenate((events, new_events))

    # Get metadata and events
    metadata, events, event_id = mne.epochs.make_metadata(
        events=events_temp, event_id=event_id, tmin=tmin, tmax=tmax,
        sfreq=raw.info["sfreq"], row_events=list(stimulus_id.keys()), keep_first=["condition"])
    
    # Initialize trial index list and trial counter
    trial_indices = []
    trial_index = 0

    s1_indices = []
    s2_indices = []

    # Iterate through events and assign trial indices
    i = 0
    while i < len(events):
        if i < len(events) - 1 and ((events[i][2] == 11 and events[i + 1][2] == 64) or (events[i][2] == 64 and events[i + 1][2] == 11)):
            # Valid pair (11 -> 64 or 64 -> 11)
            trial_indices.append(trial_index)  # First event of the pair
            trial_indices.append(trial_index)  # Second event of the pair
            trial_index += 1  # Increment trial index for the next pair
            i += 2  # Move past the pair
        else:
            # Event doesn't form a valid pair, start a new trial for the next event
            trial_indices.append(trial_index)  # This event gets its own trial
            trial_index += 1  # Increment trial index for the next event
            i += 1  # Move to the next event
    
    metadata['trial_index'] = trial_indices

    for idx, e in enumerate(events):
        if e[2] == 11:
            s1_indices.append(idx)
        elif e[2] == 64:
            s2_indices.append(idx)

    return events, metadata, s1_indices, s2_indices

In [40]:
def create_epochs_t1(tmin, tmax, raw):
    events, metadata, s1_indices, _ = get_events(raw, tmin, tmax)
    metadata = metadata.iloc[s1_indices]
    # events = events[s1_indices]
    events = [events[i] for i in s1_indices]
    
    keep_cols = ['response/t1', 'first_condition', 'trial_index']
    metadata = metadata[keep_cols]
    metadata.reset_index(drop=True, inplace=True) # This allows to preserve the trial number after the rejections performed by the mne.Epochs function
    metadata.columns = ['rt', 'condition', 'trial_index'] # More convenient names

    event_id = {'stimulus/t1': 11} # S1 will be the onset of the epochs

    # Create epochs
    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=False,
                baseline=(None, 0), preload=True, 
                verbose=True, detrend=1, on_missing = 'warn', event_repeated='drop',
                metadata=metadata, reject_by_annotation=True, reject=None)
    
    return epochs, events

In [41]:
def create_epochs_t2(tmin, tmax, raw):
    events, metadata, s1_indices, s2_indices = get_events(raw, tmin, tmax)
    metadata_2 = metadata.iloc[s1_indices]
    metadata = metadata.iloc[s2_indices]
    events = [events[i] for i in s2_indices]
    
    # Move up RTs of S2 based on short/long SOA because they are originally based on S1
    metadata_2['response/t2'] = np.where(metadata_2['first_condition']=='short', metadata_2['response/t2']-0.3, metadata_2['response/t2']-1.2)
    metadata['response/t2'] = metadata['response/t2'] - 0.024
    metadata.reset_index(drop=True, inplace=True)
    metadata_2.reset_index(drop=True, inplace=True)

    # Create a mapping from trial_index to first_condition from metadata_2 and update first_condition column in metadata only where trial_index matches
    update_map = metadata_2.set_index('trial_index')['first_condition']
    metadata['first_condition'] = metadata['trial_index'].map(update_map).fillna(metadata['first_condition'])

    update_map = metadata_2.set_index('trial_index')['response/t2']
    metadata['response/t2'] = metadata['trial_index'].map(update_map).fillna(metadata['response/t2'])
    
    keep_cols = ['response/t2', 'first_condition', 'trial_index']
    metadata = metadata[keep_cols]
    metadata.reset_index(drop=True, inplace=True) # This allows to preserve the trial number after the rejections performed by the mne.Epochs function
    metadata.columns = ['rt', 'condition', 'trial_index'] # More convenient names

    event_id = {'stimulus/t2': 64} # S2 will be the onset of the epochs

    # Create epochs
    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=False,
                baseline=(None, 0), preload=True, 
                verbose=True, detrend=1, on_missing = 'warn', event_repeated='drop',
                metadata=metadata, reject_by_annotation=True, reject=None)
    
    return epochs, events

In [42]:
def create_epochs_trial(tmin, tmax, raw):
    events, metadata, s1_indices, _ = get_events(raw, tmin, tmax)
    metadata = metadata.iloc[s1_indices]
    # events = events[s1_indices]
    events = [events[i] for i in s1_indices]
    
    keep_cols = ['response/t2', 'first_condition', 'trial_index']
    metadata = metadata[keep_cols]
    metadata.reset_index(drop=True, inplace=True) # This allows to preserve the trial number after the rejections performed by the mne.Epochs function
    metadata.columns = ['rt', 'condition', 'trial_index'] # More convenient names

    event_id = {'stimulus/t1': 11} # S1 will be the onset of the epochs

    # Create epochs
    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=False,
                baseline=(None, 0), preload=True, 
                verbose=True, detrend=1, on_missing = 'warn', event_repeated='drop',
                metadata=metadata, reject_by_annotation=True, reject=None)
    
    return epochs, events

In [240]:
def autoreject(epochs, raw):
    picks = mne.pick_types(raw.info, eeg=False, stim=False, eog=False, include=['C3', 'C4', 'POz', 'FCz'], exclude=[])
    ar = aj.AutoReject()
    
    # Fit on the first 20 epochs to save time
    ar.fit(epochs[0:20])
    epochs_ar, reject_log = ar.transform(epochs, return_log=True)

    return epochs_ar

In [241]:
# Given a tuple of ndarrays (index 0 containing event, index 1 containing time)
# Return fixed events
def merge_events_stim(stim):
    stim_new = stim[0].copy()
    mapping = {
        # stim/t1
        12: 11,
        20: 11,
        21: 11,
        22: 11,
        23: 11,
        31: 11,
        32: 11,
        # stim/t2
        65: 64,
        66: 64,
        67: 64,
        68: 64,
        128: 64,
        129: 64,
        130: 64,
        131: 64,
        132: 64,
        133: 64,
        # response/t1
        2: 1,
        3: 1,
        # response/t2
        5: 4,
        # block
        7: 6,
    }
    for k, v in mapping.items():
        stim_new[stim[0] == k] = v
    return stim_new


def find_condition(events):
    # If stim_t2 occurs within 300 or 1200ms from stim_t1, include condition
    # Otherwise exclude that stim_t1 timepoint
    new_events = []
    
    stims_t1 = events[events[:, 2] == 11, 0]
    stims_t2 = events[events[:, 2] == 64, 0]

    # Identify short/long SOA based on time between stims
    for idx, onset in enumerate(stims_t1):
        diffs = stims_t2[stims_t2 > onset] - onset
        # diffs[0] is distance (in samples?) to closest t2 stim
        if 600 < diffs[0] < 615:  # Most often 602/603
            # Long SOA
            new_events.append([onset, 0, 256])
        elif 140 < diffs[0] < 151:  # Most often 141/142/143
            # Short SOA
            new_events.append([onset, 0, 512])
        else:
            # Invalid
            new_events.append([onset, 0, 1024])
    return new_events


def get_events(raw, tmin, tmax):
    stimulus_id = {"stimulus/t1": 11, "stimulus/t2": 64}  # Stimulus trigger
    condition_id = {"condition/short": 512, "condition/long": 256}  # Condition trigger
    response_id = {"response/t2": 4, "response/t1": 1}  # Response triggers
    event_id = condition_id | stimulus_id | response_id  # All retained events

    # Correct event trigger values in raw data
    stim_idx = raw.ch_names.index("Status")
    raw._data[stim_idx] = merge_events_stim(raw["Status"])

    # IMPORTANT, set consecutive=True so not only ascending trigger values are recorded as new events
    events = mne.find_events(raw, shortest_event=1, consecutive=True)

    # find_condition adds condition trigger
    new_events = np.array(find_condition(events))
    # 1, 4, 6, 11, 16, 64

    # Append events to end of array, now has events 256/512 at same onset, but later in array
    events_tmp = np.concatenate((events, new_events))

    # Makes for every event in row_events (11/64, stim/t1 and stim/t2)
    # a row containing all other events (condition, response)
    metadata, events, event_id = mne.epochs.make_metadata(
        events=events_tmp,
        event_id=event_id,
        tmin=tmin,
        tmax=tmax,
        sfreq=raw.info["sfreq"],
        row_events=list(stimulus_id.keys()),
        keep_first=["condition"],
    )
    # keep_first causes first_condition to be set for stim/t2 based on the NEXT trial
    # if 'condition' is not 0 (condition was not set at the same time as event), set first_condition to shift(1) value (previous row)
    metadata.loc[(metadata['condition'] != 0), 'first_condition'] = metadata['first_condition'].shift(1)

    # Trial is valid if first_condition is not None AND response/t1 is not NaN AND response/t2 is not NaN, in these cases we know that there was ~300/1200ms between stimuli and responses were given in the time between stim/t1 and next stim/t1
    metadata.loc[metadata["event_name"] == "stimulus/t1", "valid"] = (
        metadata["first_condition"].notna()
        & metadata["response/t1"].notna()
        & metadata["response/t2"].notna()
    )

    # t2 'trials' are valid when the previous row was stim/t1 and it was valid, AND response/t2 is not na
    metadata.loc[metadata["event_name"] == "stimulus/t2", "valid"] = (
        (metadata["event_name"].shift(1) == "stimulus/t1")
        & (metadata["valid"].shift(1) == True)
        & (metadata["response/t2"].notna())
    )

    metadata = metadata.reset_index(drop=True)

    trial_index = 0
    # Assign trial index to each row, the same number for every stimulus/t1 that is followed by a stimulus/t2
    for row in metadata.itertuples():
        if row.event_name == "stimulus/t1":
            metadata.at[row.Index, "trial_index"] = trial_index
            trial_index += 1
        elif row.event_name == "stimulus/t2":
            if metadata.at[row.Index - 1, "event_name"] == "stimulus/t1":
                metadata.at[row.Index, "trial_index"] = trial_index - 1
            else:
                metadata.at[row.Index, "trial_index"] = trial_index
                trial_index += 1
    metadata['trial_index'] = metadata.trial_index.astype(int)

    # find stim/t1 and stim/t2 indices
    s1_indices = metadata[(metadata["event_name"] == "stimulus/t1") & (metadata["valid"] == True)].index.tolist()
    s2_indices = metadata[(metadata["event_name"] == "stimulus/t2") & (metadata["valid"] == True)].index.tolist()

    return events, metadata, s1_indices, s2_indices

In [247]:
metadata['first_condition'][:50]

0      None
1      None
2      long
3      long
4     short
5     short
6      long
7      long
8     short
9     short
10    short
11    short
12    short
13    short
14     long
15     long
16    short
17    short
18     long
19     long
20    short
21    short
22     long
23     long
24     long
25     long
26     long
27     long
28     long
29     long
30    short
31    short
32    short
33    short
34     long
35     long
36    short
37    short
38    short
39    short
40     long
41     long
42     long
43     long
44     long
45     long
46     long
47     long
48     long
49     long
Name: first_condition, dtype: object

In [242]:
def create_epochs_t1(tmin, tmax, raw):
    events, metadata, s1_indices, _ = get_events(raw, tmin, tmax)

    s1_indices = [i for i in s1_indices]

    metadata = metadata.iloc[s1_indices]
    events = [events[i] for i in s1_indices]
    keep_cols = ['response/t1', 'first_condition', 'trial_index']
    metadata = metadata[keep_cols]
    metadata.reset_index(drop=True, inplace=True) # This allows to preserve the trial number after the rejections performed by the mne.Epochs function
    metadata.columns = ['rt', 'condition', 'trial_index'] # More convenient names
    event_id = {'stimulus/t1': 11} # S1 will be the onset of the epochs
    
    # Create epochs
    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=False,
                baseline=(None, 0), preload=True, 
                verbose=True, detrend=1, on_missing = 'warn', event_repeated='drop',
                metadata=metadata, reject_by_annotation=True, reject=None)
    
    return epochs, events


In [243]:
def create_epochs_t2(tmin, tmax, raw):
    events, metadata, _, s2_indices = get_events(raw, tmin, tmax)

    s2_indices = [i for i in s2_indices]

    metadata = metadata.iloc[s2_indices]
    events = [events[i] for i in s2_indices]
    keep_cols = ['response/t2', 'first_condition', 'trial_index']
    metadata = metadata[keep_cols]
    metadata.reset_index(drop=True, inplace=True) # This allows to preserve the trial number after the rejections performed by the mne.Epochs function
    metadata.columns = ['rt', 'condition', 'trial_index'] # More convenient names
    event_id = {'stimulus/t2': 64} # S1 will be the onset of the epochs
    
    # Create epochs
    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=False,
                baseline=(None, 0), preload=True, 
                verbose=True, detrend=1, on_missing = 'warn', event_repeated='drop',
                metadata=metadata, reject_by_annotation=True, reject=None)
    
    return epochs, events

In [245]:
tmin = -0.25
tmax = 2
for subj in subj_names:
    raw = mne.io.read_raw_fif(path / f"post_ica_{subj}_raw.fif", preload=True, verbose=False)

    # stimulus/t1 > response/t1
    epochs, events = create_epochs_t1(tmin, tmax, raw)
    epochs = autoreject(epochs, raw)
    epochs.save(out_path / f"{subj}-t1-epo.fif", overwrite=True)

    # stimulus/t2 > response/t2
    epochs, events = create_epochs_t2(tmin, tmax, raw) 
    epochs = autoreject(epochs, raw)
    epochs.save(out_path / f"{subj}-t2-epo.fif", overwrite=True)

    # TODO: Implement stimulus/t1 > response/t2

Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6582 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1244 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1244 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1244 [00:00<?,       ?it/s]

Dropped 29 epochs: 63, 112, 190, 260, 315, 322, 346, 423, 428, 514, 531, 546, 561, 586, 587, 645, 650, 705, 718, 765, 769, 829, 876, 907, 977, 1012, 1040, 1067, 1108
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6582 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1244 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1244 events and 1153 original time points ...
9 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1235 [00:00<?,       ?it/s]

Dropped 22 epochs: 63, 78, 312, 315, 334, 344, 425, 510, 542, 556, 582, 583, 701, 715, 806, 871, 903, 971, 1033, 1035, 1101, 1213
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6498 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1223 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1223 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.70 and n_interpolate=32


  0%|          | Repairing epochs : 0/1223 [00:00<?,       ?it/s]

Dropped 34 epochs: 20, 86, 216, 223, 258, 264, 296, 309, 528, 529, 530, 539, 542, 554, 557, 558, 579, 626, 627, 637, 657, 669, 780, 1002, 1003, 1004, 1015, 1112, 1169, 1182, 1187, 1188, 1195, 1217
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6498 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1223 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1223 events and 1153 original time points ...
10 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1213 [00:00<?,       ?it/s]

Dropped 50 epochs: 20, 70, 78, 134, 215, 222, 257, 294, 327, 416, 435, 524, 525, 526, 540, 547, 549, 550, 552, 553, 556, 560, 575, 621, 632, 647, 652, 664, 691, 768, 773, 774, 775, 953, 977, 995, 996, 997, 1019, 1029, 1045, 1048, 1104, 1152, 1160, 1177, 1178, 1179, 1191, 1208
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6487 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64 69]
Adding metadata with 3 columns
952 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 952 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=4


  0%|          | Repairing epochs : 0/952 [00:00<?,       ?it/s]

Dropped 102 epochs: 22, 23, 58, 59, 60, 61, 63, 64, 65, 66, 67, 70, 71, 75, 76, 77, 80, 83, 85, 88, 89, 90, 94, 97, 100, 101, 102, 109, 110, 121, 122, 126, 127, 130, 132, 135, 136, 137, 138, 139, 140, 143, 144, 147, 148, 150, 151, 152, 154, 155, 157, 158, 159, 160, 162, 164, 165, 166, 167, 168, 169, 170, 174, 175, 176, 178, 180, 182, 184, 185, 191, 197, 198, 201, 205, 207, 208, 209, 210, 211, 213, 214, 216, 219, 222, 223, 224, 227, 230, 238, 240, 243, 245, 246, 248, 249, 253, 257, 258, 300, 369, 479
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6487 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64 69]
Adding metadata with 3 columns
952 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 952 events and 1153 original time points ...
7 bad epochs dr

  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.30 and n_interpolate=4


  0%|          | Repairing epochs : 0/945 [00:00<?,       ?it/s]

Dropped 319 epochs: 5, 14, 19, 22, 23, 24, 25, 26, 27, 31, 32, 34, 36, 47, 52, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 2

  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1039 [00:00<?,       ?it/s]

Dropped 16 epochs: 221, 359, 439, 441, 442, 465, 466, 467, 662, 668, 695, 697, 809, 847, 883, 891
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6480 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1038 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1038 events and 1153 original time points ...
2 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1036 [00:00<?,       ?it/s]

Dropped 16 epochs: 141, 183, 439, 441, 442, 465, 466, 467, 581, 661, 662, 694, 696, 807, 887, 924
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6474 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1240 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1240 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1240 [00:00<?,       ?it/s]

Dropped 9 epochs: 94, 610, 616, 617, 664, 928, 1045, 1136, 1137
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6474 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1240 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1240 events and 1153 original time points ...
7 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1233 [00:00<?,       ?it/s]

Dropped 10 epochs: 94, 292, 606, 610, 612, 775, 798, 923, 1130, 1131
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6480 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1254 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1254 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1254 [00:00<?,       ?it/s]

Dropped 8 epochs: 203, 309, 310, 311, 315, 317, 617, 1136
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6480 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1253 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1253 events and 1153 original time points ...
5 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1248 [00:00<?,       ?it/s]

Dropped 10 epochs: 274, 309, 310, 316, 317, 322, 333, 344, 381, 1100
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6494 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1281 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1281 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1281 [00:00<?,       ?it/s]

Dropped 21 epochs: 45, 245, 315, 580, 609, 688, 689, 743, 744, 754, 787, 812, 878, 879, 927, 932, 948, 949, 994, 1155, 1237
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6494 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1281 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1281 events and 1153 original time points ...
2 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1279 [00:00<?,       ?it/s]

Dropped 27 epochs: 245, 315, 466, 580, 609, 688, 704, 743, 744, 754, 786, 787, 811, 877, 879, 927, 932, 948, 992, 993, 1043, 1098, 1153, 1215, 1235, 1236, 1276
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6436 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1077 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1077 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.70 and n_interpolate=32


  0%|          | Repairing epochs : 0/1077 [00:00<?,       ?it/s]

Dropped 160 epochs: 38, 42, 43, 45, 46, 47, 53, 55, 58, 59, 64, 65, 68, 74, 76, 79, 82, 85, 86, 87, 89, 91, 96, 97, 102, 105, 117, 118, 127, 135, 155, 163, 178, 186, 197, 198, 211, 213, 229, 237, 258, 282, 295, 300, 306, 312, 316, 363, 364, 365, 368, 369, 370, 373, 375, 377, 378, 382, 383, 384, 385, 391, 396, 398, 402, 408, 411, 415, 417, 423, 430, 431, 445, 459, 522, 524, 526, 534, 543, 553, 554, 555, 556, 557, 560, 561, 566, 571, 572, 578, 584, 586, 629, 631, 636, 640, 653, 655, 668, 677, 686, 704, 774, 775, 792, 821, 822, 823, 825, 835, 836, 837, 839, 842, 844, 845, 852, 856, 857, 862, 866, 868, 870, 871, 875, 876, 879, 881, 886, 893, 901, 905, 911, 913, 914, 915, 917, 918, 919, 921, 923, 924, 927, 928, 929, 941, 947, 948, 950, 953, 954, 955, 957, 966, 1021, 1048, 1049, 1068, 1075, 1076
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6436 events foun

  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1070 [00:00<?,       ?it/s]

Dropped 209 epochs: 38, 42, 43, 45, 46, 47, 50, 54, 55, 57, 58, 64, 65, 67, 68, 69, 74, 76, 79, 82, 83, 85, 86, 87, 88, 89, 90, 91, 95, 96, 97, 101, 102, 105, 110, 115, 117, 118, 135, 136, 140, 154, 159, 162, 163, 170, 178, 186, 197, 198, 211, 215, 230, 258, 282, 286, 295, 300, 306, 307, 316, 342, 346, 353, 355, 363, 364, 365, 367, 369, 370, 372, 375, 377, 380, 381, 382, 383, 384, 385, 390, 391, 396, 397, 398, 405, 407, 408, 410, 411, 412, 415, 422, 429, 430, 457, 459, 485, 516, 517, 519, 521, 523, 529, 531, 540, 550, 551, 552, 553, 554, 555, 556, 557, 558, 562, 568, 572, 573, 574, 575, 581, 582, 583, 604, 626, 628, 633, 637, 638, 652, 674, 676, 700, 733, 743, 744, 770, 771, 788, 791, 806, 811, 818, 819, 821, 831, 832, 833, 838, 840, 841, 842, 843, 845, 847, 848, 849, 852, 853, 858, 862, 867, 869, 870, 872, 873, 876, 881, 886, 889, 891, 896, 900, 901, 906, 908, 909, 910, 912, 913, 916, 918, 919, 922, 923, 924, 926, 927, 931, 936, 937, 942, 943, 944, 948, 949, 950, 952, 957, 961, 1015, 

  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1232 [00:00<?,       ?it/s]

Dropped 32 epochs: 201, 303, 320, 360, 445, 473, 492, 498, 516, 559, 560, 564, 619, 717, 762, 804, 805, 852, 853, 911, 912, 913, 914, 931, 953, 1019, 1020, 1043, 1062, 1218, 1219, 1221
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6470 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1231 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1231 events and 1153 original time points ...
5 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1226 [00:00<?,       ?it/s]

Dropped 36 epochs: 200, 301, 318, 358, 414, 442, 489, 495, 512, 555, 556, 559, 614, 712, 758, 800, 801, 820, 839, 848, 849, 906, 907, 908, 909, 925, 948, 1013, 1014, 1015, 1038, 1056, 1057, 1211, 1212, 1213
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6489 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1187 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1187 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.40 and n_interpolate=4


  0%|          | Repairing epochs : 0/1187 [00:00<?,       ?it/s]

Dropped 46 epochs: 0, 1, 2, 3, 4, 5, 6, 9, 10, 11, 12, 15, 16, 17, 18, 19, 20, 21, 23, 27, 39, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 72, 73, 80, 86, 87, 88, 91, 93, 99, 100, 102, 1169, 1170
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6489 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1187 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1187 events and 1153 original time points ...
3 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1184 [00:00<?,       ?it/s]

Dropped 32 epochs: 0, 1, 2, 3, 5, 9, 10, 11, 12, 15, 16, 17, 18, 19, 20, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 86, 87, 88, 91, 1166, 1167
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
5399 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 18 64]
Adding metadata with 3 columns
952 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 952 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/952 [00:00<?,       ?it/s]

Dropped 124 epochs: 52, 82, 132, 159, 172, 188, 210, 220, 282, 289, 306, 316, 324, 331, 332, 338, 343, 368, 369, 410, 423, 426, 435, 436, 439, 446, 448, 451, 455, 456, 457, 461, 465, 466, 467, 468, 469, 470, 471, 473, 477, 481, 510, 551, 588, 589, 590, 599, 603, 604, 618, 635, 647, 648, 655, 656, 657, 661, 663, 664, 665, 666, 669, 671, 673, 674, 677, 680, 681, 683, 685, 686, 690, 692, 697, 698, 700, 701, 714, 716, 719, 722, 723, 725, 728, 731, 732, 736, 737, 739, 740, 750, 753, 766, 768, 769, 771, 774, 775, 779, 780, 786, 791, 793, 801, 802, 811, 814, 825, 829, 833, 851, 857, 860, 867, 868, 883, 884, 889, 891, 893, 927, 928, 930
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
5399 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 18 64]
Adding metadata with 3 columns
952 matching events found
Setting baseline interval to [-0.25, 0.0] s
Appl

  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/950 [00:00<?,       ?it/s]

Dropped 210 epochs: 52, 82, 130, 132, 159, 168, 169, 171, 172, 173, 174, 188, 190, 207, 211, 257, 289, 298, 300, 306, 311, 316, 318, 331, 332, 338, 343, 368, 371, 406, 409, 410, 421, 423, 433, 435, 439, 448, 449, 450, 451, 452, 454, 455, 456, 457, 458, 459, 461, 464, 465, 466, 467, 468, 469, 470, 471, 473, 476, 477, 482, 487, 488, 490, 510, 519, 522, 525, 554, 557, 558, 565, 570, 580, 585, 587, 588, 589, 598, 602, 617, 618, 621, 623, 628, 634, 645, 646, 652, 653, 654, 655, 657, 659, 660, 661, 662, 663, 664, 666, 667, 669, 670, 671, 672, 673, 674, 675, 678, 679, 680, 681, 682, 683, 687, 688, 690, 695, 697, 698, 699, 701, 702, 703, 705, 708, 712, 714, 717, 719, 720, 722, 726, 727, 728, 729, 730, 731, 733, 734, 735, 737, 738, 739, 744, 747, 748, 749, 751, 754, 762, 764, 765, 766, 767, 771, 772, 773, 774, 777, 778, 783, 786, 788, 789, 791, 800, 803, 807, 808, 809, 812, 813, 816, 817, 826, 827, 828, 830, 831, 833, 835, 836, 841, 849, 850, 853, 854, 855, 857, 865, 868, 874, 875, 878, 880, 88

  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1196 [00:00<?,       ?it/s]

Dropped 82 epochs: 45, 112, 162, 183, 225, 226, 238, 239, 252, 253, 255, 256, 268, 285, 319, 326, 336, 342, 355, 363, 369, 380, 402, 420, 422, 433, 455, 481, 483, 488, 503, 584, 588, 593, 599, 644, 690, 691, 710, 718, 719, 722, 723, 730, 756, 759, 773, 779, 785, 795, 797, 822, 825, 833, 858, 859, 860, 870, 871, 873, 886, 904, 908, 914, 929, 937, 938, 956, 959, 990, 1053, 1054, 1055, 1056, 1068, 1085, 1096, 1125, 1140, 1153, 1156, 1172
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6494 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1195 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1195 events and 1153 original time points ...
2 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1193 [00:00<?,       ?it/s]

Dropped 76 epochs: 45, 83, 111, 161, 224, 225, 237, 238, 252, 254, 284, 293, 317, 334, 340, 344, 361, 376, 377, 400, 418, 420, 430, 479, 486, 501, 582, 587, 597, 619, 642, 673, 688, 689, 716, 720, 727, 753, 754, 757, 770, 771, 777, 780, 783, 792, 793, 795, 805, 820, 822, 823, 856, 868, 869, 871, 884, 902, 906, 907, 912, 927, 935, 936, 951, 954, 957, 1051, 1053, 1066, 1082, 1094, 1123, 1137, 1153, 1189
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6458 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1211 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1211 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1211 [00:00<?,       ?it/s]

Dropped 42 epochs: 66, 177, 364, 417, 428, 436, 438, 448, 456, 459, 460, 469, 471, 476, 479, 481, 492, 494, 499, 528, 531, 532, 534, 536, 538, 558, 559, 577, 582, 604, 606, 666, 782, 783, 784, 785, 852, 963, 1100, 1150, 1166, 1190
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6458 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1211 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1211 events and 1153 original time points ...
3 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1208 [00:00<?,       ?it/s]

Dropped 49 epochs: 40, 66, 176, 235, 358, 362, 420, 425, 430, 434, 443, 454, 457, 458, 467, 474, 477, 479, 483, 490, 492, 497, 526, 528, 529, 530, 532, 534, 537, 540, 558, 566, 575, 578, 580, 594, 595, 601, 662, 779, 780, 781, 798, 849, 960, 1097, 1147, 1163, 1175
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6488 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1247 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1247 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1247 [00:00<?,       ?it/s]

Dropped 53 epochs: 28, 199, 407, 512, 513, 514, 515, 516, 517, 518, 519, 520, 521, 522, 523, 524, 525, 526, 527, 528, 529, 530, 531, 532, 533, 534, 535, 536, 537, 538, 539, 644, 796, 826, 828, 829, 830, 831, 883, 884, 885, 897, 1016, 1027, 1028, 1029, 1035, 1105, 1143, 1147, 1154, 1157, 1220
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6488 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1247 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1247 events and 1153 original time points ...
8 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1239 [00:00<?,       ?it/s]

Dropped 70 epochs: 20, 21, 28, 199, 200, 289, 302, 313, 404, 456, 508, 509, 510, 511, 512, 513, 514, 515, 516, 517, 518, 519, 520, 521, 522, 523, 524, 525, 526, 527, 528, 529, 530, 531, 532, 533, 534, 535, 584, 639, 686, 791, 820, 821, 822, 823, 824, 852, 876, 877, 878, 879, 891, 949, 1020, 1021, 1022, 1028, 1030, 1044, 1099, 1135, 1139, 1145, 1146, 1149, 1192, 1211, 1215, 1237
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6481 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1239 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1239 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1239 [00:00<?,       ?it/s]

Dropped 66 epochs: 31, 34, 36, 57, 58, 61, 64, 73, 74, 76, 137, 180, 181, 207, 254, 255, 256, 259, 261, 264, 265, 266, 267, 268, 270, 271, 275, 283, 289, 411, 413, 418, 419, 420, 476, 477, 486, 493, 535, 536, 537, 621, 646, 647, 648, 651, 659, 660, 667, 668, 669, 670, 671, 673, 957, 1027, 1028, 1029, 1030, 1053, 1055, 1070, 1118, 1175, 1176, 1177
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6481 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1239 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1239 events and 1153 original time points ...
4 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.70 and n_interpolate=32


  0%|          | Repairing epochs : 0/1235 [00:00<?,       ?it/s]

Dropped 61 epochs: 30, 36, 61, 64, 73, 76, 137, 179, 180, 207, 254, 256, 259, 261, 263, 264, 266, 267, 268, 269, 270, 274, 283, 289, 412, 413, 419, 420, 476, 477, 486, 493, 534, 536, 619, 620, 644, 645, 646, 650, 658, 659, 666, 667, 668, 669, 672, 952, 953, 1023, 1024, 1025, 1026, 1048, 1049, 1066, 1114, 1171, 1172, 1173, 1215
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6470 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1239 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1239 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.70 and n_interpolate=32


  0%|          | Repairing epochs : 0/1239 [00:00<?,       ?it/s]

Dropped 35 epochs: 37, 90, 92, 171, 172, 191, 202, 205, 285, 286, 304, 358, 490, 491, 562, 563, 610, 613, 615, 616, 628, 654, 665, 714, 716, 725, 818, 864, 910, 923, 1069, 1134, 1172, 1207, 1208
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6470 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1239 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1239 events and 1153 original time points ...
7 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.70 and n_interpolate=32


  0%|          | Repairing epochs : 0/1232 [00:00<?,       ?it/s]

Dropped 35 epochs: 37, 90, 92, 168, 171, 191, 202, 205, 285, 286, 303, 356, 357, 488, 489, 559, 560, 609, 612, 623, 624, 661, 711, 720, 812, 858, 904, 906, 916, 1061, 1127, 1128, 1131, 1164, 1200
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6473 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1220 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1220 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1220 [00:00<?,       ?it/s]

Dropped 15 epochs: 96, 284, 309, 343, 503, 505, 584, 645, 672, 683, 785, 786, 821, 1152, 1171
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6473 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1220 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1220 events and 1153 original time points ...
6 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1214 [00:00<?,       ?it/s]

Dropped 23 epochs: 70, 96, 97, 283, 306, 307, 308, 341, 342, 501, 502, 503, 582, 643, 670, 681, 749, 782, 817, 851, 1146, 1164, 1203
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6495 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1231 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1231 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1231 [00:00<?,       ?it/s]

Dropped 24 epochs: 23, 97, 240, 291, 351, 352, 372, 411, 430, 445, 451, 534, 658, 684, 692, 738, 776, 777, 859, 1001, 1085, 1125, 1148, 1169
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6495 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1231 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1231 events and 1153 original time points ...
7 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1224 [00:00<?,       ?it/s]

Dropped 17 epochs: 23, 97, 238, 349, 370, 426, 427, 597, 654, 688, 733, 771, 772, 853, 995, 1079, 1119
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6473 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1229 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1229 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1229 [00:00<?,       ?it/s]

Dropped 45 epochs: 23, 28, 43, 45, 55, 56, 57, 61, 62, 63, 64, 67, 79, 81, 85, 91, 127, 129, 140, 141, 148, 152, 155, 157, 158, 159, 172, 184, 192, 219, 239, 240, 250, 266, 290, 303, 351, 465, 473, 494, 580, 593, 690, 795, 1012
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6473 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1229 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1229 events and 1153 original time points ...
4 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1225 [00:00<?,       ?it/s]

Dropped 48 epochs: 23, 28, 42, 43, 45, 55, 56, 57, 61, 62, 63, 67, 79, 81, 84, 91, 94, 127, 129, 132, 140, 147, 152, 155, 157, 158, 159, 172, 184, 217, 218, 237, 238, 239, 248, 268, 288, 301, 315, 333, 349, 463, 591, 633, 688, 792, 874, 1008
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6522 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
57 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 57 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/57 [00:00<?,       ?it/s]

Dropped 14 epochs: 11, 18, 26, 31, 33, 34, 37, 39, 42, 43, 44, 45, 46, 47
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6522 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
57 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 57 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.80 and n_interpolate=32


  0%|          | Repairing epochs : 0/57 [00:00<?,       ?it/s]

Dropped 8 epochs: 18, 26, 31, 33, 34, 44, 45, 46
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6492 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
58 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 58 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.70 and n_interpolate=32


  0%|          | Repairing epochs : 0/58 [00:00<?,       ?it/s]

Dropped 16 epochs: 18, 24, 31, 36, 38, 39, 40, 43, 44, 49, 50, 51, 53, 54, 56, 57
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6492 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
58 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 58 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.50 and n_interpolate=4


  0%|          | Repairing epochs : 0/58 [00:00<?,       ?it/s]

Dropped 23 epochs: 18, 24, 26, 29, 31, 35, 36, 37, 38, 39, 40, 42, 43, 44, 48, 49, 50, 51, 52, 54, 55, 56, 57
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
5944 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1152 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1152 events and 1153 original time points ...
0 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1152 [00:00<?,       ?it/s]

Dropped 108 epochs: 7, 12, 25, 35, 38, 95, 99, 106, 158, 163, 171, 199, 203, 219, 257, 264, 278, 289, 309, 315, 319, 320, 321, 349, 353, 359, 369, 371, 389, 418, 434, 464, 469, 488, 517, 521, 587, 610, 621, 625, 626, 627, 629, 631, 642, 645, 688, 691, 731, 743, 793, 794, 815, 837, 842, 852, 856, 902, 910, 917, 918, 919, 920, 921, 1004, 1018, 1021, 1022, 1023, 1029, 1041, 1046, 1049, 1050, 1051, 1066, 1068, 1071, 1072, 1082, 1093, 1094, 1095, 1096, 1097, 1098, 1099, 1100, 1103, 1104, 1114, 1115, 1117, 1119, 1120, 1121, 1122, 1124, 1130, 1133, 1140, 1141, 1142, 1143, 1144, 1146, 1147, 1151
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
5944 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1151 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using 

  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.60 and n_interpolate=32


  0%|          | Repairing epochs : 0/1144 [00:00<?,       ?it/s]

Dropped 114 epochs: 7, 12, 25, 35, 38, 57, 62, 95, 98, 99, 106, 158, 163, 171, 175, 198, 202, 263, 277, 287, 288, 308, 314, 317, 319, 320, 323, 349, 352, 365, 370, 388, 416, 432, 452, 462, 477, 486, 487, 511, 515, 518, 584, 605, 612, 616, 620, 621, 622, 640, 659, 670, 683, 707, 726, 788, 789, 810, 830, 831, 835, 836, 846, 896, 904, 906, 907, 911, 912, 913, 914, 915, 1010, 1011, 1014, 1015, 1016, 1020, 1022, 1039, 1042, 1043, 1063, 1064, 1065, 1074, 1086, 1087, 1088, 1089, 1091, 1092, 1093, 1095, 1096, 1099, 1103, 1107, 1108, 1112, 1113, 1114, 1115, 1117, 1122, 1123, 1126, 1132, 1133, 1134, 1135, 1136, 1137, 1139
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6484 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1245 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline corre

  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.80 and n_interpolate=32


  0%|          | Repairing epochs : 0/1245 [00:00<?,       ?it/s]

Dropped 36 epochs: 9, 18, 23, 24, 35, 50, 56, 58, 67, 89, 107, 113, 114, 115, 147, 158, 193, 230, 329, 347, 353, 369, 426, 555, 616, 639, 690, 727, 759, 788, 846, 851, 854, 895, 916, 917
Trigger channel Status has a non-zero initial value of {initial_value} (consider using initial_event=True to detect this event)
Removing orphaned offset at the beginning of the file.
6484 events found on stim channel Status
Event IDs: [ 1  4  6 11 16 64]
Adding metadata with 3 columns
1244 matching events found
Setting baseline interval to [-0.25, 0.0] s
Applying baseline correction (mode: mean)
Using data from preloaded Raw for 1244 events and 1153 original time points ...
9 bad epochs dropped
Running autoreject on ch_type=eeg


  0%|          | Creating augmented epochs : 0/64 [00:00<?,       ?it/s]

  0%|          | Computing thresholds ... : 0/64 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | n_interp : 0/3 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]

  0%|          | Repairing epochs : 0/20 [00:00<?,       ?it/s]

  0%|          | Fold : 0/10 [00:00<?,       ?it/s]





Estimated consensus=0.70 and n_interpolate=32


  0%|          | Repairing epochs : 0/1235 [00:00<?,       ?it/s]

Dropped 60 epochs: 17, 22, 23, 24, 35, 49, 50, 55, 56, 58, 66, 82, 89, 106, 113, 114, 115, 116, 120, 121, 142, 147, 157, 169, 174, 193, 205, 229, 300, 305, 327, 345, 350, 351, 362, 367, 375, 393, 410, 414, 422, 423, 500, 611, 634, 663, 685, 686, 721, 722, 754, 783, 839, 844, 847, 865, 888, 910, 969, 984


### DEBUGGING

In [117]:
cnt = 0
for subj in subj_names:
    print(subj)
    raw = mne.io.read_raw_fif(path / f"post_ica_{subj}_raw.fif", preload=True, verbose=False)
    print(raw['stim'])
    print(raw.ch_names)
    stim_idx = raw.ch_names.index('Status')
    res = merge_events_stim(raw['Status'])
    raw._data[stim_idx] = res[0]
    print(res[0][:,414450:414480])
    print(raw['stim'][0][:,414450:414480])
    print(raw['Status'][0][:,414450:414480])
    # Consecutive? Assumes that trial onset is first, probably very unlikely that two numbers in sequence will be non-zero so doesn't matter?
    # Q LEENDERT: 
    events = mne.find_events(raw, shortest_event=1)
    events_cons = mne.find_events(raw, shortest_event=1, consecutive=True)
    events = merge_events(events)
    events_cons = merge_events(events_cons)
    # print(events[:,2][:100])
    # print(events_cons[:,2][:100])
    # print(np.unique(events[:,2], return_counts=True))
    # print(np.unique(events_cons[:,2], return_counts=True))
    
    for i in range(1, len(events)):
        if events[i-1, 2] == 64 and events[i, 2] == 64:
            print(events[i-1], events[i])
            cnt += 1
    break
    # for i in range(0, events_cons.shape[0], 100):
    #     print(i)
    #     events_subset = events[i:i+100]
    #     cons_subset = events_cons[i:i+100]
    #     if (events_subset != cons_subset).any():
    #         print(events_subset[:,2])
    #         print(cons_subset[:,2])
    #         print(events_subset[:,0])
    #         print(cons_subset[:,0])
    #         break
    
print(cnt)

VP10
(array([[65536., 65536., 65536., ...,     0.,     0.,     0.]]), array([0.00000000e+00, 1.95312500e-03, 3.90625000e-03, ...,
       3.94999414e+03, 3.94999609e+03, 3.94999805e+03]))
['Fp1', 'AF7', 'AF3', 'F1', 'F3', 'F5', 'F7', 'FT7', 'FC5', 'FC3', 'FC1', 'C1', 'C3', 'C5', 'T7', 'TP7', 'CP5', 'CP3', 'CP1', 'P1', 'P3', 'P5', 'P7', 'P9', 'PO7', 'PO3', 'O1', 'Iz', 'Oz', 'POz', 'Pz', 'CPz', 'Fpz', 'Fp2', 'AF8', 'AF4', 'AFz', 'Fz', 'F2', 'F4', 'F6', 'F8', 'FT8', 'FC6', 'FC4', 'FC2', 'FCz', 'Cz', 'C2', 'C4', 'C6', 'T8', 'TP8', 'CP6', 'CP4', 'CP2', 'P2', 'P4', 'P6', 'P8', 'P10', 'PO8', 'PO4', 'O2', 'Status', 'eog_h', 'eog_v']
[[ 0.  0.  0.  0.  0. 64. 64. 64. 64. 64. 64. 64. 64. 64. 64. 64.  1.  1.
   1.  1.  1.  1.  0.  0.  0.  0.  0.  0.  0.  0.]]
[[ 0.  0.  0.  0.  0. 64. 64. 64. 64. 64. 64. 64. 64. 64. 64. 64.  1.  1.
   1.  1.  1.  1.  0.  0.  0.  0.  0.  0.  0.  0.]]
[[ 0.  0.  0.  0.  0. 64. 64. 64. 64. 64. 64. 64. 64. 64. 64. 64.  1.  1.
   1.  1.  1.  1.  0.  0.  0.  0.  0.  0. 

In [91]:
raw['stim'][0]

array([[65536., 65536., 65536., ...,     0.,     0.,     0.]])

In [97]:
raw['stim'][0][:,414450:414480]

array([[ 0.,  0.,  0.,  0.,  0., 64., 64., 64., 64., 64., 64., 67., 67.,
        67., 67., 67.,  3.,  3.,  3.,  3.,  3.,  3.,  0.,  0.,  0.,  0.,
         0.,  0.,  0.,  0.]])

244 TOTAL occurences where 64 follows 64, 153 if consecutive='increasing' (stim t2), what does this mean?
Likely cases where 64 is followed by not 64 (but a value that is corrected to 64 in merge_events)
solution: merge_events BEFORE find_events

1/2/3 + = 14.392 (all responses t1)
4/5 + = 14.529 (all responses t2)
6 = new block?
11/12/21/22/31/32 += stim t1
130/129/128/67/66/64 (should be 64/128, failure when transmitting int?) = stim t2
16: fixation cross
Normal trial = 16-11-64-1-4/16-11-1-64-4
ALWAYS 16 (fixation) 11 (stim t1) > 1 (resp t1) OR 64 (stim t2) > 4
Not all response events were recorded, participant was too slow? Should not count trials where this is the case

In [33]:
raw['Status'][0]

array([[65536., 65536., 65536., ...,     0.,     0.,     0.]])

In [49]:
raw.info

<Info | 11 non-empty values
 bads: []
 ch_names: Fp1, AF7, AF3, F1, F3, F5, F7, FT7, FC5, FC3, FC1, C1, C3, C5, ...
 chs: 64 EEG, 1 Stimulus, 2 EOG
 custom_ref_applied: True
 dig: 67 items (3 Cardinal, 64 EEG)
 file_id: 4 items (dict)
 highpass: 1.0 Hz
 lowpass: 100.0 Hz
 meas_date: 2016-11-28 15:09:22 UTC
 meas_id: 4 items (dict)
 nchan: 67
 projs: []
 sfreq: 512.0 Hz
>

In [40]:
np.unique(raw['Status'][0].astype(int), return_counts=True)
# 1/2/3 + = 14.392 (all responses t1)
# 4/5 + = 14.529 (all responses t2)
# 6 = new block?
# 11/12/21/22/31/32 += stim t1
# 130/129/128/67/66/64 (should be 64/128, failure when transmitting int?) = stim t2
# 16: fixation cross

(array([    0,     1,     2,     3,     4,     5,     6,    11,    12,
           16,    21,    22,    31,    32,    64,    66,    67,   128,
          129,   130, 65536]),
 array([1949969,    4647,    4990,    4755,    7263,    7266,     143,
           2451,    2384,   14466,    2394,    2418,    2395,    2418,
           7174,       6,       5,    7115,      15,       6,     120],
       dtype=int64))

In [52]:
raw.info

<Info | 11 non-empty values
 bads: []
 ch_names: Fp1, AF7, AF3, F1, F3, F5, F7, FT7, FC5, FC3, FC1, C1, C3, C5, ...
 chs: 64 EEG, 1 Stimulus, 2 EOG
 custom_ref_applied: True
 dig: 67 items (3 Cardinal, 64 EEG)
 file_id: 4 items (dict)
 highpass: 1.0 Hz
 lowpass: 100.0 Hz
 meas_date: 2016-11-28 15:09:22 UTC
 meas_id: 4 items (dict)
 nchan: 67
 projs: []
 sfreq: 512.0 Hz
>

In [ ]:
tmin = -0.25
tmax = 2

for subj in subj_names:
    # Skip those subjects because the data is not valid
    if subj == 'VP6' or subj == 'VP7':
        continue

    raw = mne.io.read_raw_fif(path+"Post ICA/post_ica_%s_raw.fif"%subj, preload=True)
    # epoch = mne.read_epochs(epoch_data_path+subj+'.fif')

    # ================= STIMULUS 1 =================
    epochs, events = create_epochs_t1(tmin, tmax, raw)
    epochs = autoreject(epochs, raw)
    epochs.save(path+'Stimlocked_fixed_index/%s_epo_t1.fif'%subj, overwrite=True)

    # # ================= STIMULUS 2 =================
    epochs, events = create_epochs_t2(tmin, tmax, raw)
    epochs = autoreject(epochs, raw)
    epochs.save(path+'Stimlocked_fixed_index/%s_epo_t2.fif'%subj, overwrite=True)

    # ================= WHOLE TRIALS =================
    # epochs, events = create_epochs_trial(tmin, tmax, raw)
    # epochs = autoreject(epochs, raw)
    # epochs.save(path+'Stimlocked_fixed_index/%s_epo_trial.fif'%subj, overwrite=True)